apps/api/src/app/partner-integrations/usecases/update-vercel-configuration/update-vercel-configuration.usecase.ts

```diff
@@ -1,405 +1,172 @@
-      }, []);
+      const { organizationId, configurationId } = command;
 
-      const envKeys = await this.getEnvKeys(organizationIds);
+      const configuration = await this.findPartnerConfigurationDetail({
+        organizationId,
+        configurationId,
+      });
 
-      const configurationDetails = await this.getVercelProjectsUsecase.getVercelConfiguration(command.environmentId, {
-        configurationId: command.configurationId,
+      const organizationsWithConfiguration = await this.organizationRepository.findPartnerConfigurationDetails({
         userId: command.userId,
+        configurationId: command.configurationId,
       });
 
-      await this.updateBridgeUrl(command.environmentId, projectIds[0], configurationDetails.accessToken);
-
-      const { newAndUpdatedProjectIds, removedProjectIds } = await this.getUpdatedAndRemovedProjectIds(
-        command,
-        projectIds
-      );
-
-      const { addProjectIds, updateProjectDetails, removedProjectDetails } = await this.getVercelProjects(
-        configurationDetails.accessToken,
-        configurationDetails.teamId,
-        newAndUpdatedProjectIds,
-        removedProjectIds
+      const allOldProjectIds = [
+        ...new Set(
+          organizationsWithConfiguration.reduce<string[]>((acc, org) => {
+            return acc.concat(org.partnerConfigurations?.[0].projectIds || []);
+          }, [])
+        ),
+      ];
+
+      const vercelLinkedProjects = await this.getVercelLinkedProjects(
+        configuration.accessToken,
+        configuration.teamId,
+        allOldProjectIds
       );
 
       await this.removeEnvironmentVariables({
-        removedProjectDetails,
-        teamId: configurationDetails.teamId,
-        token: configurationDetails.accessToken,
+        vercelLinkedProjects,
+        teamId: configuration.teamId,
+        token: configuration.accessToken,
       });
 
-      const mappedProjectData = this.mapProjectKeys(envKeys, command.data, {
-        updateProjectDetails,
-        addProjectIds,
-      });
-
-      for (const key of Object.keys(mappedProjectData)) {
-        await this.setEnvironmentVariables({
-          clientKey: mappedProjectData[key].clientKey,
-          privateKey: mappedProjectData[key].privateKey,
-          projectIds: mappedProjectData[key].addProjectIds,
-          teamId: configurationDetails.teamId,
-          token: configurationDetails.accessToken,
-        });
-        await this.updateEnvironmentVariables({
-          clientKey: mappedProjectData[key].clientKey,
-          privateKey: mappedProjectData[key].privateKey,
-          projectDetails: mappedProjectData[key].updateProjectDetails,
-          teamId: configurationDetails.teamId,
-          token: configurationDetails.accessToken,
-        });
-      }
-
-      await this.saveProjectIds(command);
+      await this.completeVercelIntegration.execute(
+        CompleteVercelIntegrationCommand.create({
+          ...command,
+        })
+      );
```

Comment:
Reuse the complete usecase that will create all configurations and env variables once again. No need of complex logic that was here before.